WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingViewModel.swift

```diff
@@ -10,11 +11,18 @@ class StoreOnboardingViewModel: ObservableObject {
         case loading
         /// Shows a list of onboarding tasks
         case loaded(rows: [StoreOnboardingTaskViewModel])
+        /// When the request fails and there is no previously loaded local data
+        case failed
     }
 
     @Published private(set) var isRedacted: Bool = false
     @Published private(set) var taskViewModels: [StoreOnboardingTaskViewModel] = []
 
+    /// Emits when there are no tasks available for display after reload.
+    /// i.e. When (request failed && No previously loaded local data available)
+    ///
+    let noTasksAvailableForDisplay = PassthroughSubject<Void, Never>()
```

Comment:
nit: generally, we want to abstract the observable in the interface to be `AnyPublisher` as in most of the `PassthroughSubject` use cases in the app where they have an accompanying `AnyPublisher` for the public interface. this way, how the publisher is implemented (pass-through or current value) is not exposed and can be changed anytime.